apps/swirl-docs/src/components/ComponentPreview/CodePreview.tsx

```diff
@@ -0,0 +1,134 @@
+}
+
+export const CodePreview: FunctionComponent<CodePreviewProps> = ({
+  component,
+}) => {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+  const [code, setCode] = useState<string>("");
+
+  const generateCode = useCallback(() => {
+    if (component?.tag) {
+      const el = document.createElement(component.tag);
+
+      component.props.forEach((prop: Prop) => {
+        if (prop.default) {
+          const propDefaultValue = prop.default as string;
+
+          const cleanedPropValue = propDefaultValue.replace(/"/g, "");
+
+          el.setAttribute(prop.name, cleanedPropValue);
+        }
+      });
+
+      if (component?.innerHtml) {
+        el.innerHTML = component.innerHtml;
+      }
+
+      return prettier.format(el.outerHTML, {
+        parser: "html",
+        plugins: [prettierHTML],
+      });
+    }
+
+    return "";
+  }, [component]);
+
+  useEffect(() => {
+    setCode(generateCode());
+  }, [generateCode]);
+
+  return (
+    <NoSsr>
+      <div
+        className={classNames(
+          "relative w-full bg-[#24292E]  rounded-lg mb-10 overflow-auto",
+          {
+            "h-[240px] max-h-[240px] overflow-hidden": !isExpanded,
+            "h-full min-h-[240px]": isExpanded,
+          }
+        )}
+      >
+        <div className="flex items-center justify-between bg-[#21201E] h-12 m-2 rounded-lg p-4 ">
+          <div className="flex items-center justify-between">
+            <Link href="#">
+              <a className="flex justify-center items-center text-[#F2F2F2] text-base font-medium mr-4">
+                npm package
+                <FlipIconOpenInNew className="ml-1" size={16} />
+              </a>
+            </Link>
+            <CodeSandboxButton code={code} />
+          </div>
+          <CopyToClipboard
+            text={code}
+            onCopy={() => {
+              setIsCopied(true);
+              setTimeout(() => {
+                setIsCopied(false);
+              }, 2000);
+            }}
+          >
+            <button className="flex justify-center items-center text-[#F2F2F2] text-base font-medium">
+              {isCopied ? "copied!" : "copy"}
+              {isCopied ? (
+                <FlipIconCheckStrong size={16} className="ml-1" />
+              ) : (
+                <FlipIconCopy size={16} className="ml-1" />
+              )}
+            </button>
+          </CopyToClipboard>
+        </div>
+        <SyntaxHighlighter
+          wrapLines
+          wrapLongLines
+          style={dark}
+          customStyle={{
+            backgroundColor: "#24292E",
+            padding: "0.5rem 1.5rem 4rem 1.5rem", // place for the expand button
+            height: "100%",
+          }}
+        >
+          {code}
+        </SyntaxHighlighter>
+        <div className="absolute bottom-0 flex justify-center items-center w-full h-12  bg-[#24292E]">
+          <button
```

Comment:
**question:** Is it possible to hide this, when the code is not overflowing?

Reply:
yeah just thought about that aswell, when i checked it on the big screen here in the office